### PR TITLE
refactor: Fix ragg2 for arm targets by using the correct gcc

### PR DESCRIPTION
**Pull request by nitanmarcel**
- [?] Mark this if you consider it ready to merge (if you are ok for now to not have the same done for clang, or llvm-gcc yeah.
- [X] I've added tests (optional)
- [ ] I wrote some lines in the [book](https://github.com/radareorg/radare2book) (optional)

**Description**

When using ragg2 with an arm(32/64) arch it will still try to use the default compiler (in my case gcc) instead of using the cross one failing to compile succesfully.

I have no idea what's wrong with clang. `clang -target arm-linux-gnueabihf` works and creates arm binaries but when using the clang command generated by `ragg2` doesn't.

```
⋊> ~/s/example clang -target arm-linux-gnueabihf a.c                                                                                                                                   17:50:45
⋊> ~/s/example file a.out                                                                                                                                                              17:50:52
a.out: ELF 32-bit LSB pie executable, ARM, EABI5 version 1 (SYSV), dynamically linked, interpreter /lib/ld-linux-armhf.so.3, BuildID[sha1]=c744cd56ef48cd1305c5090cbaa3f3e09f9b9390, for GNU/Linux 3.2.0, not stripped 
```

```
⋊> ~/s/example clang -target arm-linux-gnueabihf -fPIC -fPIE -pie -fpic -nostartfiles -fno-stack-protector -nostdinc -include '/usr/local/include/libr/sflib'/'linux-arm-32'/sflib.h -z execstack -fomit-frame-pointer -finline-functions -fno-zero-initialized-in-bss -o 'a.c.tmp' -S 'a.c'
clang: warning: -z execstack: 'linker' input unused [-Wunused-command-line-argument]
clang: warning: argument unused during compilation: '-pie' [-Wunused-command-line-argument]
clang: warning: argument unused during compilation: '-nostartfiles' [-Wunused-command-line-argument]
In file included from <built-in>:1:
/usr/local/include/libr/sflib/linux-arm-32/sflib.h:38:15: error: invalid output constraint '=a' in asm
   38 | static inline _sfsyscall6(void *,mmap, void *,start, size_t,length, int,prot , int,flags, int,fd, off_t,offset)
      |               ^
/usr/local/include/libr/sflib/linux-arm-32/sfsyscall.h:168:7: note: expanded from macro '_sfsyscall6'
  168 |                                 : "=a" (__res)                                  \
      |                                   ^
In file included from <built-in>:1:
/usr/local/include/libr/sflib/linux-arm-32/sflib.h:53:15: error: invalid output constraint '=a' in asm
   53 | static inline _sfsyscall0( int, sched_yield )
......
```

### CODE REVIEW

File: libr/egg/egg_cfile.c [hunk 1]
```diff
@@ -77,7 +95,7 @@ static struct cEnv_t* r_egg_cfile_set_cEnv(const char *arch, const char *os, int
 		return NULL;
 	}
 
-	if (!(cEnv->CC = r_egg_cfile_getCompiler())) {
+	if (!(cEnv->CC = r_egg_cfile_getCompiler(arch))) {
```
Comment (trufae) [inline libr/egg/egg_cfile.c:98]:
```suggestion
	if (!(cEnv->CC = r_egg_cfile_getCompiler (arch))) {
```

File: libr/egg/egg_cfile.c [hunk 2]
```diff
@@ -16,18 +16,36 @@ struct cEnv_t {
...
+  	}
+
+  	if (strcmp(arch, "arm32")) {
+    		compiler = "arm-linux-gnueabihf-gcc";
+    		compiler_path = r_file_path(compiler);
+  	}
+
+  	if (compiler_path) {
+    		free(compiler_path);
+    		return strdup(compiler);
```
Comment (trufae) [inline libr/egg/egg_cfile.c:44]:
```suggestion
    		return strdup (compiler);
```

File: libr/egg/egg_cfile.c [hunk 3]
```diff
@@ -16,18 +16,36 @@ struct cEnv_t {
...
+    		compiler_path = r_file_path(compiler);
+  	}
+
+  	if (strcmp(arch, "arm32")) {
+    		compiler = "arm-linux-gnueabihf-gcc";
+    		compiler_path = r_file_path(compiler);
+  	}
+
+  	if (compiler_path) {
+    		free(compiler_path);
```
Comment (trufae) [inline libr/egg/egg_cfile.c:43]:
```suggestion
    		free (compiler_path);
```

**Review by trufae — CHANGES_REQUESTED** [review]
Spaces before ( in function calls pls

**Comment by nitanmarcel** [comment]
> Spaces before ( in function calls pls

Done

**Comment by trufae** [comment]
Uhm i can squash but if you can rewrite the commit messages using the guidelines rules it will be better because those two commits have separate logics.

- start in uppercase (remove the refactor keyword)
- Use ##ragg doubletat so the commit message will be taken by the release notes script

**Comment by nitanmarcel** [comment]
> Collaborator

Done

**Pull request merged**